setup assistant crashes if there are no remotes

### Steps to reproduce

```sh
rm -rf test || true
mkdir test
cd test
git init
git commit --allow-empty -m initial
git-town hack test

# the setup assistant opens, hit enter on all questions
```

### Command output

```shell
Aliased commands: (all)
Main branch: main

Development Remote

Which remote should Git Town use
for development?

Typically that's the "origin" remote.

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/git-town/git-town/v21/internal/cli/dialog/dialogcomponents/list.List[...].SelectedEntry(...)
	/Users/kevlar/d/git-town/internal/cli/dialog/dialogcomponents/list/list.go:181
github.com/git-town/git-town/v21/internal/cli/dialog/dialogcomponents/list.List[...].SelectedData(...)
	/Users/kevlar/d/git-town/internal/cli/dialog/dialogcomponents/list/list.go:177
github.com/git-town/git-town/v21/internal/cli/dialog/dialogcomponents.RadioList[...]({0x1054282c0, 0x0?, 0x104d45e00?}, 0x14000490090?, {0x1048aa4bf?, 0x12}, {0x1048d4edb, 0x5b}, {{0x14000243560}, {0x1054282c0, ...}, ...}, ...)
	/Users/kevlar/d/git-town/internal/cli/dialog/dialogcomponents/radiolist.go:25 +0x24c
github.com/git-town/git-town/v21/internal/cli/dialog.DevRemote({0x1054282c0, 0x0, 0x1?}, {{0x0}, {{0x14000243560}, {0x1054282c0, 0x0, 0x0}, 0x0}, {0x0}})
	/Users/kevlar/d/git-town/internal/cli/dialog/dev_remote.go:61 +0x400
github.com/git-town/git-town/v21/internal/setup.enterDevRemote({{0x104d3f0d8, 0x1400000c9d8}, {{0x1400021a570, {0x0}, {0x0}, {0x0}, 0x1400021a5a0, {0x0}, {0x0}, {0x0}, ...}, ...}, ...})
	/Users/kevlar/d/git-town/internal/setup/enter.go:366 +0x9c
github.com/git-town/git-town/v21/internal/setup.Enter({{0x104d3f0d8, 0x1400000c9d8}, {{0x1400021a570, {0x0}, {0x0}, {0x0}, 0x1400021a5a0, {0x0}, {0x0}, {0x0}, ...}, ...}, ...})
	/Users/kevlar/d/git-town/internal/setup/enter.go:42 +0x3b4
github.com/git-town/git-town/v21/internal/validate.Config({{0x104d3f0d8, 0x1400000c9d8}, {0x1400028c7b0, 0x1, 0x1}, 0x1400028c990, {0x140002a4260, 0x1, 0x1}, {0x1400021aae0, ...}, ...})
	/Users/kevlar/d/git-town/internal/validate/config.go:38 +0x248
github.com/git-town/git-town/v21/internal/cmd.determineAppendData({_, _, {_}, _, _, {_, _}}, {{0x104d3f0d8, 0x1400000c9d8}, {0x1400000fa98}, ...})
	/Users/kevlar/d/git-town/internal/cmd/append.go:293 +0x113c
github.com/git-town/git-town/v21/internal/cmd.executeAppend({{0x16bc7f8b8, 0x5}, 0x0, {0x1400021a570, {0x0}, {0x0}, {0x0}, 0x1400021a5a0, {0x0}, {0x0}, ...}, ...})
	/Users/kevlar/d/git-town/internal/cmd/append.go:152 +0x200
github.com/git-town/git-town/v21/internal/cmd.appendCmd.func1(0x140001fa608, {0x1400002ee20, 0x1, 0x1048a13d4?})
	/Users/kevlar/d/git-town/internal/cmd/append.go:107 +0x384
github.com/spf13/cobra.(*Command).execute(0x140001fa608, {0x1400002edf0, 0x1, 0x1})
	/Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x140001fa008)
	/Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1071
github.com/git-town/git-town/v21/internal/cmd.Execute()
	/Users/kevlar/d/git-town/internal/cmd/doc.go:49 +0x434
main.main()
	/Users/kevlar/d/git-town/main.go:18 +0x28
[main 3917614] commit
 1 file changed, 1 insertion(+)
 create mode 100644 child
```

### Additional information

The error occurs when selecting the remote. This repo has no remote.

--- a/internal/cli/dialog/dev_remote.go
+++ b/internal/cli/dialog/dev_remote.go
@@ -25,6 +25,9 @@ Typically that's the "origin" remote.
 )
 
 func DevRemote(remotes gitdomain.Remotes, args Args[gitdomain.Remote]) (Option[gitdomain.Remote], dialogdomain.Exit, error) {
+	if len(remotes) == 0 {
+		return None[gitdomain.Remote](), false, nil
+	}
 	options := list.Entries[Option[gitdomain.Remote]]{}
 	global, hasGlobal := args.Global.Get()
 	if hasGlobal {

